src/azure-cli-core/azure/cli/core/commands/client_factory.py

```diff
@@ -137,15 +137,23 @@ def _get_mgmt_service_client(cli_ctx,
                              api_version=None,
                              base_url_bound=True,
                              resource=None,
+                             scopes=None,
                              sdk_profile=None,
                              aux_subscriptions=None,
                              aux_tenants=None,
                              **kwargs):
```

Comment:
@qianwens @jiasli I have an interesting finding here. It turned out that if I use the `--credential-scopes` parameter in the autorest generation, it automatically adds my scope into the generated client library:

```
class PublicApiClientConfiguration(Configuration):

    def __init__(
        self,
        credential,  # type: "TokenCredential"
        **kwargs  # type: Any
    ):
        # type: (...) -> None
        if credential is None:
            raise ValueError("Parameter 'credential' must not be None.")
        super(PublicApiClientConfiguration, self).__init__(**kwargs)

        self.credential = credential
        self.credential_scopes = ['https://ostechnology.onmicrosoft.com/azsphereapi/user_impersonation']
        self.credential_scopes.extend(kwargs.pop('credential_scopes', []))
        kwargs.setdefault('sdk_moniker', 'sphere-publicapi/{}'.format(VERSION))
        self._configure(**kwargs)
```

This means that extending this function is not really necessary if the generated SDK already has the scope internally as in my example above.

I still believe that it does make sense to extend this API with the `scopes` since the `resource` is already there. I just wanted to draw your attention on this. What is your view about this?

Reply:
This approach makes the SDK lose the capability to customize the scope from outside the SDK. For example, ARM in `AzureCloud` has https://github.com/Azure/azure-cli/blob/81ff7bfad67383a9d8aba2e2e0bd5a7457b0f8b9/src/azure-cli-core/azure/cli/core/cloud.py#L250 

while `AzureChinaCloud` has https://github.com/Azure/azure-cli/blob/81ff7bfad67383a9d8aba2e2e0bd5a7457b0f8b9/src/azure-cli-core/azure/cli/core/cloud.py#L281

Reply:
This is correct. However, if the SDK is generate with `azure-arm` flag (which is should be), it adds `https://management.core.windows.net/` as the default `credential_scopes` item. For example, in storage, as a random example: https://github.com/Azure/azure-sdk-for-python/blob/ba66869562f94559b3f167bc8d3a733157b67573/sdk/storage/azure-mgmt-storage/azure/mgmt/storage/_configuration.py#L46

Reply:
@jsntcy, let's make sure we pass the correct `credential_scopes` to Track 2 SDKs.